crates/bevy_app/src/plugin.rs

```diff
@@ -12,4 +12,4 @@ pub trait Plugin: Any + Send + Sync {
     }
 }
 
-pub type CreatePlugin = unsafe fn() -> *mut dyn Plugin;
+pub type CreatePlugin = fn() -> *mut dyn Plugin;
```

Comment:
You are correct that without documentation explaining why this is unsafe, these two are strictly equivalent

However, that doesn't mean that either version is actually correct. In particular, the only usage of `CreatePlugin` currently assumes that it returns a valid plugin instance (as a note, we currently mix up the heaps in `dynamically_load_plugin` so I don't think it can ever be safe, but that's someone else's problem). With the unsafe marker, these conditions are at least implied, which is better (but still not great). Imo, `CreatePlugin` should be an implementation detail of `bevy_dynamic_plugin`, and should not exist as a type alias at all.

Reply:
Alright, maybe I'll get to removing it. Not sure.